Lib/test/test_queue.py

```diff
@@ -10,6 +10,16 @@
 from test.support import import_helper
 from test.support import threading_helper
 
+import typing as t
+
+if t.TYPE_CHECKING:
+    import queue
+
+    parent_class = unittest.TestCase
+
+else:
+    parent_class = object
```

Comment:
I'm not excited about using `typing` here. We're not running mypy or pyright over the stdlib or the test suite, so inevitably at some point it will be wrong and an incorrect type annotation is worse than no type annotation.

It's also a bit of a code smell that you have to use `TYPE_CHECKING` at all to get around `unittest`'s test discovery for the mixin class. (Maybe there are patterns you can borrow from elsewhere? I'm sure this is a pretty common pattern, but nothing else in the test suite uses `TYPE_CHECKING` AFAICT.)